### PR TITLE
Feature/comments web UI

### PR DESCRIPTION
**Pull request by lups2000**
#### Description

Added the possibility to add a comment as a column in the flow table. Then, added a section in the flow details panel where we can see and modify the comment. I modified the backend as well since we were not supporting the modification of a comment. I hope the way I did it is correct :)

closes : #6708 

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by lups2000** [comment]
if you like it, I can then update the tests. Let me know :) thank you

File: web/src/js/components/FlowTable/FlowColumns.tsx [hunk 1]
```diff
@@ -291,6 +292,13 @@ export const quickactions: FlowColumn = ({ flow }) => {
     );
 };
 
+export const comment: FlowColumn = ({ flow }) => {
+    const text = getComment(flow);
+    return <td className="col-comment">{text ? text : "..."}</td>;
```
Comment (mhils) [inline web/src/js/components/FlowTable/FlowColumns.tsx:297]: "..." sounds more like "too long to display" than no comment. Do we just want to leave it empty maybe?

File: web/src/js/components/FlowView.tsx [hunk 2]
```diff
@@ -9,13 +9,14 @@ import Connection from "./FlowView/Connection";
...
 import WebSocket from "./FlowView/WebSocket";
-
+import Comment from "./FlowView/Comment";
 import { selectTab } from "../ducks/ui/flow";
 import { useAppDispatch, useAppSelector } from "../ducks";
 import { Flow } from "../flow";
 import classnames from "classnames";
 import TcpMessages from "./FlowView/TcpMessages";
 import UdpMessages from "./FlowView/UdpMessages";
+import { tab } from "@testing-library/user-event/dist/tab";
```
Comment (mhils) [inline web/src/js/components/FlowView.tsx:19]: leftover

**Review by mhils — COMMENTED** [review]
This looks good, thanks!

File: web/src/js/components/FlowView.tsx [hunk 3]
```diff
@@ -58,6 +60,7 @@ export function tabsForFlow(flow: Flow): string[] {
     if (flow.error) tabs.push("error");
     tabs.push("connection");
     tabs.push("timing");
+    tabs.push("comment");
```
[the remark this answers is not in the document]
Comment (lups2000) [reply web/src/js/components/FlowView.tsx:63]: Another solution I had was to change the "Timing" tabs into "Info" or "Details" and in that section put the timing content and the comment (below the timing)

[the remark this answers is not in the document]
Comment (mhils) [reply web/src/js/components/FlowView.tsx:63]: That's what I was thinking too. Both works for me I guess!

[the remark this answers is not in the document]
Comment (lups2000) [reply web/src/js/components/FlowView.tsx:63]: yes agree, I can change that. A tab just for a comment is probably too much

File: web/src/js/flow/utils.ts [hunk 4]
```diff
@@ -191,3 +191,7 @@ export const getTotalSize = (flow: Flow): number => {
 export const canReplay = (flow: Flow): boolean => {
     return flow.type === "http" && !flow.websocket;
 };
+
+export const getComment = (flow: Flow): string => {
+    return flow.comment;
+};
```
[the remark this answers is not in the document]
Comment (lups2000) [reply web/src/js/flow/utils.ts:197]: okay yes thanks

File: web/src/js/components/FlowTable/FlowColumns.tsx (hunk 1 above)
Comment (lups2000) [reply web/src/js/components/FlowTable/FlowColumns.tsx:297]: yes makes sense

**Comment by lups2000** [comment]
@mhils I made a change in the `app.css` file, basically adding the style for the comment column in the flow table. In general, as stated in the README, this file shouldn't be pushed, but in this case can I push my changes?

**Comment by mhils** [comment]
> @mhils I made a change in the `app.css` file, basically adding the style for the comment column in the flow table. In general, as stated in the README, this file shouldn't be pushed, but in this case can I push my changes?

app.css is autogenerated from the less files. So you want to modify the less files in https://github.com/mitmproxy/mitmproxy/tree/main/web/src/css and keep app.css as-is. It will be automatically updated on the next release. :)

**Comment by lups2000** [comment]
> > @mhils I made a change in the `app.css` file, basically adding the style for the comment column in the flow table. In general, as stated in the README, this file shouldn't be pushed, but in this case can I push my changes?
> 
> app.css is autogenerated from the less files. So you want to modify the less files in https://github.com/mitmproxy/mitmproxy/tree/main/web/src/css and keep app.css as-is. It will be automatically updated on the next release. :)

ahhh I see now, thank you !

**Comment by lups2000** [comment]
it seems there are issues with the coverage. I don't understand why `onEditDone` isn't tested by `ValueEditorSpec.tsx`

**Comment by lups2000** [comment]
@mhils Is there a reason why the folder `editors` is being ignored in the coverage report (line 13 `jest.config.js` )? Due to this, I believe I may not be able to pass the codecov/patch test. I am just missing 2 lines :(

**Comment by mhils** [comment]
a) looks like we can remove the editors exclusion.
b) failing coverage CI is not an absolute blocker. I'll try to take a closer look at the PR later today

**Review by mhils — COMMENTED** [review]
Looks good, thanks!

The more I look at it the more I want to merge it with the Timings pane, but let's resist the urge and get this in first. :)

**Comment by lups2000** [comment]
> Looks good, thanks!
> 
> The more I look at it the more I want to merge it with the Timings pane, but let's resist the urge and get this in first. :)

Alright :) I'll do it in the future, I promise ahah. For the coverage ? Is it fine ?

**Review by mhils — APPROVED** [review]

**Pull request merged**